src/test/java/greencity/services/impl/UserServiceImplTest.java

```diff
@@ -0,0 +1,76 @@
+package greencity.services.impl;
+
+import static org.junit.Assert.*;
+
+import greencity.GreenCityApplication;
+import greencity.entity.User;
+import greencity.entity.enums.ROLE;
+import greencity.service.UserService;
+import java.time.LocalDateTime;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.junit4.SpringRunner;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(classes = GreenCityApplication.class)
+@DirtiesContext(classMode = DirtiesContext.ClassMode.AFTER_EACH_TEST_METHOD)
+public class UserServiceImplTest {
+
+    @Autowired private UserService userService;
+
+    @Test
+    public void blockUserTest() {
+        User user =
+                User.builder()
+                        .firstName("test")
+                        .lastName("test")
+                        .email("test@gmail.com")
+                        .role(ROLE.USER_ROLE)
+                        .isBlocked(false)
+                        .lastVisit(LocalDateTime.now())
+                        .dateOfRegistration(LocalDateTime.now())
+                        .build();
+        userService.save(user);
```

Comment:
Better use mockito instead of saving test user in the database.
When you are writing tests for service layer, you need to mock all the other classes (such as repository), so you are sure every time when this test fails it is only because of code in service class (but not because of db failture and so on)

Reply:
resolved